### PR TITLE
fix RectangleZoom selection vertices

### LINKED ISSUE #2673: log scale breaks zoom controls
Log scale breaks the zoom controls. As soon as I initialize `Axis` with `xscale=log10`, zoom controls break.
Screenshot 2023-02-11 at 8 57 08 PM

### PR DESCRIPTION
**Pull request by EdsterG**
# Description

Fixes #2673

## Type of change

Delete options that do not apply:

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] Added an entry in NEWS.md (for new features and breaking changes)
- [ ] Added or changed relevant sections in the documentation
- [ ] Added unit tests for new algorithms, conversion methods, etc.
- [ ] Added reference image tests for new plotting functions, recipes, visual options, etc.

### CODE REVIEW

File: src/makielayout/interactions.jl [hunk 1]
```diff
@@ -122,11 +122,12 @@ function _chosen_limits(rz, ax)
     return r
 end
 
-function _selection_vertices(ax_scene, outer, inner)
+function _selection_vertices(ax, outer, inner)
```
Comment (asinghvi17) [inline src/makielayout/interactions.jl:125]:
```suggestion
function _selection_vertices(ax_scene, outer, inner)

File: src/makielayout/types.jl [hunk 2]
```diff
@@ -468,7 +468,7 @@ end
 
 function RectangleZoom(f::Function, ax::Axis; kw...)
     r = RectangleZoom(f; kw...)
-    selection_vertices = lift(_selection_vertices, Observable(ax.scene), ax.finallimits, r.rectnode)
+    selection_vertices = lift(_selection_vertices, Observable(ax), ax.finallimits, r.rectnode)
```
Comment (asinghvi17) [inline src/makielayout/types.jl:471]:
```suggestion
    selection_vertices = lift(_selection_vertices, Observable(ax.scene), ax.finallimits, r.rectnode)
```

File: src/makielayout/interactions.jl [hunk 3]
```diff
@@ -122,11 +122,12 @@ function _chosen_limits(rz, ax)
...
     _clamp(p, plow, phigh) = Point2f(clamp(p[1], plow[1], phigh[1]), clamp(p[2], plow[2], phigh[2]))
-    proj(point) = project(ax_scene, point) .+ minimum(ax_scene.px_area[])
-    outer = positivize(outer)
-    inner = positivize(inner)
+    proj(point) = project(ax.scene, point) .+ minimum(ax.scene.px_area[])
+    transf = Makie.transform_func(ax)
```
Comment (asinghvi17) [inline src/makielayout/interactions.jl:128]:
```suggestion
    proj(point) = project(ax_scene, point) .+ minimum(ax_scene.px_area[])
    transf = Makie.transform_func(ax_scene)

**Review by asinghvi17 — COMMENTED** [review]
Thanks for the PR!  One thing - no need to pass the axis to the `_selection_vertices` function, since you can get the transform function from the Scene.

```julia
julia> Makie.current_axis() |> Makie.transform_func
(identity, identity)

julia> Makie.current_axis().scene |> Makie.transform_func
(identity, identity)
```

**Comment by EdsterG** [comment]
Thanks for the suggestion, I updated the PR.

**Review by asinghvi17 — APPROVED** [review]
Looks good to me!

https://user-images.githubusercontent.com/32143268/218465184-7a63bb9e-03e5-4172-bc28-f0f68809ca04.mov

**Pull request merged**